R/learner_randomPlantedForest_classif_rpf.R

```diff
@@ -0,0 +1,130 @@
+#' @title Classification Random Planted Forest Learner
+#' @author jemus42
+#' @name mlr_learners_classif.rpf
+#'
+#' @description
+#' Random Planted Forest: A directly interpretable tree ensemble.
+#'
+#' Calls [randomPlantedForest::rpf()] from 'randomPlantedForest'.
+#'
+#' @section Custom mlr3 defaults:
+#' - `loss`:
+#'   - Actual default: `"L2"`.
+#'   - Adjusted default: `"exponential"`.
+#'   - Reason for change: Using `"L2"` (or `"L1"`) loss does not guarantee predictions are valid
+#'     probabilities and more akin to the linear predictor of a GLM.
+#' - `max_interaction`:
+#'   - This hyperparameter can alternatively be set via `max_interaction_ratio` as
+#'     `max_interaction = max(ceiling(max_interaction_ratio * n_features), 1)`,
+#'     and lastly `max_interaction = min(max_interaction, max_interaction_limit)`,
+#'     where `max_interaction_limit` is a second additional hyperparameter defaulting to `30`,
+#'     since more deeply nested interactions are considered impractical.
+#'     This ensures that the value `max_interaction` passed to `rpf` never exceeds
+#'     `min(n_features, max_interaction_limit)`.
+#'     This is analogous to `mtry.ratio` in [`classif.ranger`][mlr3learners::mlr_learners_classif.ranger], with
+#'     `max_interaction_limit` adding to `n_features` as an additional constraint.
+#'
+#'
+#' @section Installation:
+#' `remotes::install_github("PlantedML/randomPlantedForest")`
+#'
+#' @templateVar id classif.rpf
+#' @template learner
+#'
+#' @references
+#' `r format_bib("hiabu_2023")`
+#'
+#' @template seealso_learner
+#' @template example
+#' @export
+LearnerClassifRandomPlantedForest = R6Class("LearnerClassifRandomPlantedForest",
+  inherit = LearnerClassif,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      param_set = ps(
+        max_interaction = p_int(lower = 0, upper = Inf, default = 1, tags = "train"),
+        max_interaction_ratio = p_dbl(lower = 0, upper = 1, default = 0, tags = "train"),
+        max_interaction_limit = p_int(lower = 1, upper = Inf, default = 30, tags = "train"),
```

Comment:
for custom parameters (`max_interaction_ratio` and `max_interaction_limit`) it is better to not use defaults but initial values + add the `"required"` tag to the parameter. This ensures that you never have to deal with the parameter being `NULL`. 

I will add this to the tutorial

Reply:
Ah true. Then we need to check `if (is.null(...))` when accessing the parameter in `.train()` and `.predict()` I think